CONTRIBUTING.md

```diff
@@ -6,13 +6,75 @@ Microsoft Authentication Library (MSAL) for .NET welcomes new contributors.  Thi
+
+- You found an existing issue with a `help-wanted` label or discussed with the team and agreed on adding a new issue with that label.
+- You posted a high-level description of how the changes will be implemented and received a positive acknowledgement from the team before getting too committed to the approach or investing too much effort in implementing it.
+- You added test coverage following existing patterns within the codebase.
+- Your code matches the existing syntax conventions within the codebase.
+- Your PR is small, focused, and avoids making unrelated changes.
+
+If your pull request contains any of the below, it's less likely to be merged.
+
+- Changes that break backward compatibility
+- Changes that are only wanted by one person or company. Changes need to benefit a large enough proportion of developers using our auth libraries.
+- Changes that add entirely new feature areas without prior agreement
+- Changes that are mostly about refactoring existing code or code style
+
+Very large PRs that would take hours to review (remember, we're trying to help lots of people at once). For larger work areas, please discuss with us to find ways of breaking it down into smaller, incremental pieces that can go into separate PRs.
+
+### During a pull request review
+
+A core contributor will review your pull request and provide feedback. To ensure that there is not a large backlog of inactive PRs, the pull request will be marked as stale after two weeks of no activity. After another four days, it will be closed.
+
+### Submitting a pull request
+
+If you're not sure how to create a pull request, read GitHub's [About pull requests
+](https://help.github.com/articles/using-pull-requests) article. Make sure the repository can build and all tests pass. Familiarize yourself with the project workflow and our coding conventions. The coding style and general engineering guidelines are published on the Engineering guidelines page.
+
+### Ensuring test coverage
+
+- Tests need to be provided for every bug and feature that is completed.
+  - Unit tests must cover all new aspects of the code.
+- Before and after performance and stress tests results are successfully evaluated - no regressions are allowed.
+- Performance and stress tests are extended as relevant.
+
+Note that you won't be able to run integration tests locally because they connect to Azure Key Vault to fetch some test users and passwords.
+
+### Pull request feedback
+
+Your pull request will now go through extensive checks by the subject matter experts on our team. Please be patient; we have hundreds of pull requests across all of our repositories. Update your pull request according to feedback until it is approved by one of the team members.
+
+### Merging a pull request
+
+When your pull request has had all feedback addressed, it has been signed off by one or more reviewers with commit access, and all checks are green, we will commit it.
```

Comment:
This is a bit inaccurate since we don't run CI on forks. Idk if we need to rephrase.

Reply:
@pmaytak I didn't know that! 😁 How would you like to see it rephrased?

Reply:
It actually applies to all of our GitHub repos. I've reworded this section.